packages/integration-sdk-runtime/src/storage/FileSystemGraphObjectStore/indices.ts

```diff
@@ -4,61 +4,49 @@ import {
   GraphObjectIteratee,
   IntegrationError,
 } from '@jupiterone/integration-sdk-core';
-import { FlushedEntityData, FlushedRelationshipData } from '../types';
-import {
-  readJsonFromPath,
-  walkDirectory,
-  WalkDirectoryIterateeInput,
-} from '../../fileSystem';
+import { readJsonFromPath, WalkDirectoryIterateeInput } from '../../fileSystem';
 
 import { buildIndexDirectoryPath } from './path';
+import { iterateParsedGraphFiles } from '../..';
 
 interface IterateIndexInput<GraphObject> {
   type: string;
   iteratee: GraphObjectIteratee<GraphObject>;
 }
 
-export async function iterateEntityTypeIndex<T extends Entity = Entity>({
+interface BaseIterateIndexInput<GraphObject>
+  extends IterateIndexInput<GraphObject> {
+  collectionType: 'entities' | 'relationships';
+}
+
+async function iterateTypeIndex<T extends Entity | Relationship>({
   type,
+  collectionType,
   iteratee,
-}: IterateIndexInput<T>) {
+}: BaseIterateIndexInput<T>) {
   const path = buildIndexDirectoryPath({
-    collectionType: 'entities',
+    collectionType,
     type,
   });
 
-  await walkDirectory({
```

Comment:
The `iterateParsedGraphFiles` function was introduced in a recent PR and acts as a replacement for all of this code.

Reply:
💯   Thanks!  I should've caught that when we introduced iterateParsedGraphFiles.

Reply:
sweeet